index.bs

```diff
@@ -4900,6 +4995,57 @@ as candidates to be employed in a [=registration ceremony=].
 : Authenticator extension output
 :: None.
 
+
+## Credential Properties Extension (credProps) ## {#sctn-authenticator-credential-properties-extension}
+
+This extension allows [=[WRPS]=] to query additional properties of the [=public key credential source=] created during a 
+[=registration ceremony=].
+
+: Extension identifier
+:: `credProps`
+
+: Operation applicability
+:: [=registration extension|Registration=]
+
+: Client extension input
+:: The Boolean value [TRUE] to indicate that this extension is requested by the [=[RP]=].
+    <xmp class="idl">
+    partial dictionary AuthenticationExtensionsClientInputs {
+        boolean credProps;
+    };
+    </xmp>
+
+: Client extension processing
+:: None, except creating the authenticator extension input from the client extension input.
+
+: Client extension output
+:: Returns the credential properties as discovered by the client, if any. For example, a client is especially likely
+    to have knowledge about properties of credentials stored in [=platform authenticators=]. 
+    The client MAY omit any properties at its discretion, and SHOULD omit any properties that would be duplicated in the
+    authenticator extension output. This dictionary is defined with one credential property 
+    {{CredentialPropertiesOutput/rk}} to indicate  whether or not the returned {{PublicKeyCredential}} is a [=resident credential=].
+
+
+    <xmp class="idl">
+    dictionary CredentialPropertiesOutput {
+        boolean rk;
+    };
+
+    partial dictionary AuthenticationExtensionsClientOutputs {
+        CredentialPropertiesOutput credProps;
+    };
+    </xmp>
```

Comment:
I'm thinking we only really need to say this much about this extension given it's presently narrow purpose and single return value:

: Client extension output
:: This [=registration extension=] returns the newly-created [=credential=]'s properties to the calling [=[RP]=] [=web application=], if known by the [=client=]. For example, the [=client=] knows the parameters passed to the [=authenticatorMakeCredential=] operation and whether the operation was successful or not. The [=client=] can then use this knowledge to report to the caller.

At this time, one [=credential property=] is defined, {{CredentialPropertiesOutput/rk}}, which indicates whether or not the returned {{PublicKeyCredential}} is a [=resident credential=].

     <xmp class="idl">
    dictionary CredentialPropertiesOutput {
        boolean rk;
    };

     partial dictionary AuthenticationExtensionsClientOutputs {
        CredentialPropertiesOutput credProps;
    };
    </xmp>

Reply:
I have made a few small updates, and would like you to take another look. Keep in mind that I don't want to preclude the idea that authenticators themselves (in future) could actually return this extension data - not just the client. 

There is also not a current definition for `[=credential property=]`.

Reply:
OK, well, I'd also add the "At this time, ..." sentence to define what "rk" means, otherwise I fear it won't be clear to the reader.  

>   I don't want to preclude the idea that authenticators themselves (in future) could actually return this extension data - not just the client.

Hm.. I do not understand...  "Authenticator extension output" below says that the authnr returns the credProps, yes?

> There is also not a current definition for [=credential property=].

correct, we should create one.  I've suggested one up above.

Reply:
Ok - I obviously missed a bunch of the earlier comments. I've made the changes you've suggested, but left in this sentence:

```
The client MAY omit any properties at its discretion, and SHOULD omit any 
    properties that would be duplicated in the authenticator extension output.
```

The reason for that sentence is that if an authenticator itself reported on the rk property, the client would not need to and should not duplicate it. That's what I meant when I said that authenticators themselves could return the extension data. So might the client ... so the idea was to instruct the client not to duplicate extension parameters that an authenticator supplies.

Reply:
Ok, thanks, silly me, I think I understand now.   

Given that authnr extn outputs are distinct from client extn outputs (tho one sure needs to read this spec carefully to figure that out),  there's a few things we need to fix and further clarify in specifying this extension.

1.  I'm assuming your intent is for the client to report on whether the authn created a resident key in the event that the authnr does not support the extension and does not itself return an output value.  

2. The authnr extn output is a CBOR-encoded value that we need to express in prose and/or CDDL, rather than webIDL.  We do not have any defined means for translating from WebIDL-expressed data structures to CBOR, so we need to express the authnr extn output here in CBOR/CDDL.  E.g., see the authnr extn output for the "exts" and "txAuthSimple" extensions. 

Thoughts?

I'm working on a rewrite of this extension section which addresses the above. I'll submit it as a pull request to your sbweeden/webauthn repo later today (I hope).

Reply:
Yes, that's the idea. Thanks for working on the PR - I'll check it out when available.